js/src/ui/Form/VaultSelect/vaultSelect.js

```diff
@@ -0,0 +1,103 @@
+// Copyright 2015-2017 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import React, { Component, PropTypes } from 'react';
+import { FormattedMessage } from 'react-intl';
+
+import VaultSelector from '~/modals/VaultSelector';
+import VaultStore from '~/views/Vaults/store';
+
+import InputAddress from '../InputAddress';
+
+export default class VaultSelect extends Component {
+  static contextTypes = {
+    api: PropTypes.object.isRequired
+  };
+
+  static propTypes = {
+    onSelect: PropTypes.func.isRequired,
+    value: PropTypes.string.isRequired,
+    vaultStore: PropTypes.object
+  };
+
+  state = {
+    isOpen: false
+  };
+
+  vaultStore = this.props.vaultStore || VaultStore.get(this.context.api);
```

Comment:
Wouldn't it be better to choose whether it should be passed as prop or not? I find it a bit confusing